contracts/multihop/src/tests/swap.rs

```diff
@@ -157,6 +157,482 @@ fn swap_three_equal_pools_no_fees() {
     assert_eq!(tokens[3].balance(&recipient), 50i128);
 }
 
+#[test]
+fn swap_single_pool_no_fees() {
+    let env = Env::default();
+
+    let admin = Address::random(&env);
+    let user = Address::random(&env);
+
+    let mut token1 = deploy_token_contract(&env, &admin);
+    let mut token2 = deploy_token_contract(&env, &admin);
+
+    env.mock_all_auths();
+    env.budget().reset_unlimited();
+
+    let mut tokens = [&mut token1, &mut token2];
+    tokens.sort_by(|a, b| a.address.cmp(&b.address));
+
+    tokens[0].mint(&user, &10_000_000i128);
+    tokens[1].mint(&user, &10_000_000i128);
+
+    // 1. deploy factory
+    let factory_addr = deploy_factory_contract(&env, admin.clone());
+    let factory_client = factory::Client::new(&env, &factory_addr);
+
+    factory_client.initialize(&admin.clone());
```

Comment:
This will be a second.

Reply:
fixed with `deploy_and_mint_tokens` as per example